2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Current SDK version is 0.18.7
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Configure stats pid to 15068
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Loading settings from /root/.config/wandb/settings
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Loading settings from /tmp/pycharm_project_785/MulT-EHR/wandb/settings
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Loading settings from environment variables: {}
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Applying setup settings: {'mode': 'online', '_disable_service': None}
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Inferring run settings from compute environment: {'program_relpath': 'main.py', 'program_abspath': '/tmp/pycharm_project_785/MulT-EHR/main.py', 'program': '/tmp/pycharm_project_785/MulT-EHR/main.py'}
2024-11-16 16:29:24,730 INFO    MainThread:15068 [wandb_setup.py:_flush():79] Applying login settings: {}
2024-11-16 16:29:24,731 INFO    MainThread:15068 [wandb_init.py:_log_setup():533] Logging user logs to /tmp/pycharm_project_785/MulT-EHR/wandb/run-20241116_162924-45dgj68u/logs/debug.log
2024-11-16 16:29:24,731 INFO    MainThread:15068 [wandb_init.py:_log_setup():534] Logging internal logs to /tmp/pycharm_project_785/MulT-EHR/wandb/run-20241116_162924-45dgj68u/logs/debug-internal.log
2024-11-16 16:29:24,731 INFO    MainThread:15068 [wandb_init.py:init():619] calling init triggers
2024-11-16 16:29:24,731 INFO    MainThread:15068 [wandb_init.py:init():626] wandb.init called with sweep_config: {}
config: {'name': 'HGT_MTCausal_MIMIC4_RMDL', 'train_type': 'causal-gnn', 'eval_type': 'homo-graph', 'gpu_ids': '0', 'datasets': OrderedDict([('name', 'mimic4'), ('dataset_path', './data/mimic3_objects/mimic4_dp.pkl'), ('graph_path', './data/graphs/mimic4_dp.pkl'), ('pretrained', './data/graphs/mimic4_dp_ptr.pkl'), ('labels_path', './data/graphs/mimic4_dp_labels.pkl'), ('entity_mapping', './data/graphs/mimic4_dp_entity_mapping.pkl')]), 'checkpoint': OrderedDict([('path', './checkpoints/HGT_Causal_MIMIC4_RMDL'), ('save_checkpoint_freq', 2)]), 'optimizer': OrderedDict([('opt_method', 'ADAM'), ('lr', 0.006), ('weight_decay', 0)]), 'GNN': OrderedDict([('name', 'HGT'), ('num_layers', 2), ('in_dim', 128), ('hidden_dim', 64), ('out_dim', 506), ('num_heads', 16), ('num_out_heads', 1), ('feat_drop', 0.05)]), 'train': OrderedDict([('tasks', ['mort_pred', 'los', 'drug_rec', 'readm']), ('num_epochs', 1000), ('batch_size', 4096), ('causal', True), ('reg', 5e-05), ('n_samples', 2000), ('temperature', 0.005)]), 'logging': OrderedDict([('tags', ['MIMIC4', 'MT']), ('mode', 'online')])}
2024-11-16 16:29:24,731 INFO    MainThread:15068 [wandb_init.py:init():669] starting backend
2024-11-16 16:29:24,731 INFO    MainThread:15068 [wandb_init.py:init():673] sending inform_init request
2024-11-16 16:29:24,733 INFO    MainThread:15068 [backend.py:_multiprocessing_setup():104] multiprocessing start_methods=fork,spawn,forkserver, using: spawn
2024-11-16 16:29:24,734 INFO    MainThread:15068 [wandb_init.py:init():686] backend started and connected
2024-11-16 16:29:24,741 INFO    MainThread:15068 [wandb_init.py:init():781] updated telemetry
2024-11-16 16:29:24,769 INFO    MainThread:15068 [wandb_init.py:init():814] communicating run to backend with 90.0 second timeout
2024-11-16 16:29:25,729 INFO    MainThread:15068 [wandb_init.py:init():867] starting run threads in backend
2024-11-16 16:29:25,920 INFO    MainThread:15068 [wandb_run.py:_console_start():2456] atexit reg
2024-11-16 16:29:25,920 INFO    MainThread:15068 [wandb_run.py:_redirect():2305] redirect: wrap_raw
2024-11-16 16:29:25,920 INFO    MainThread:15068 [wandb_run.py:_redirect():2370] Wrapping output streams.
2024-11-16 16:29:25,920 INFO    MainThread:15068 [wandb_run.py:_redirect():2395] Redirects installed.
2024-11-16 16:29:25,924 INFO    MainThread:15068 [wandb_init.py:init():911] run started, returning control to user process
2024-11-16 16:29:25,929 WARNING MsgRouterThr:15068 [router.py:message_loop():75] message_loop has been closed
